File: 283-move-zeroes/283-move-zeroes.go
func moveZeroes(nums []int)  {
    
	index := 0
	count := 0

	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			count++
		} else {
			nums[index] = nums[i]
			index++
		}
	}

	for i := 0; i < count; i++ {
		nums[index+i] = 0
	}
}

File: 283-move-zeroes/283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int count = 0;
        int index = 0;

        for (int i=0; i<nums.length; i++) {
            if (nums[i] == 0 ) {
                count++;
            } else {
                nums[index] = nums[i];
                index++;
            }
        }

        for (int i=0; i<count; i++) {
            nums[index+i] = 0;
        }
    }
}